### DISCUSSION TITLE
pull new image on container restart, but do not restart on a new image

### DISCUSSION THREAD

**Discussion by thomasd3** [containrrr/watchtower#1012] [Q&A] [answered]
I'm using AWS with Fargate and their container registry.
Each time a container is started, the image is pulled from the registry.

To take advantage of this, all apps have a 'suicide' command where they just terminate. AWS' Fargate will then pull the image and restart the container.

The advantage is that I can push a new image, but the user of the app decides when the app is doing the update.

Is the same possible with WatchTower where it wouldn't stop a running app when there is a new image, but if the app has to restart it would pull the image from the repository?

**Comment by simonaronsson** [comment, marked as answer]
It sounds like `--monitor-only` is the flag you're looking for. It will pull the new image,  but not restart the container.

**Reply by thomasd3** [reply]
that's exactly what I need, thanks!
